Scripts/UtilityScripts/AzTSUtility.ps1

```diff
@@ -0,0 +1,1617 @@
+
+  $appService.PossibleOutboundIpAddresses
+}
+
+function Get-AppServiceAllCurrentOutboundPublicIps()
+{
+  <#
+    .SYNOPSIS
+    This command returns a comma-delimited string of all the CURRENT public IPs for the App Service.
+    .DESCRIPTION
+    This command returns a comma-delimited string of all the CURRENT public IPs for the App Service. These are the public IPs the App Service is currently using; they are a subset of all POSSIBLE public IPs.
+    .PARAMETER SubscriptionId
+    The Azure subscription ID containing the App Service.
+    .PARAMETER ResourceGroupName
+    The Resource Group name containing the App Service.
+    .PARAMETER AppServiceName
+    The App Service name.
+    .INPUTS
+    None
+    .OUTPUTS
+    Text with the App Service's current outbound public IP addresses.
+    .EXAMPLE
+    PS> Get-AppServiceAllCurrentOutboundPublicIps -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -AppServiceName "MyAppServiceName"
+    .LINK
+    None
+#>
+
+[CmdletBinding()]
+param (
+  [Parameter(Mandatory = $true)]
+  [string]
+  $SubscriptionId,
+  [Parameter(Mandatory = $true)]
+  [string]
+  $ResourceGroupName,
+  [Parameter(Mandatory = $true)]
+  [string]
+  $AppServiceName
+)
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  $appService = Get-AzWebApp -ResourceGroupName $ResourceGroupName -Name $AppServiceName
+
+  $appService.OutboundIpAddresses
+}
+
+function Set-KeyVaultPublicNetworkAccessEnabledForMe()
+{
+  <#
+    .SYNOPSIS
+    This command updates a Key Vault to enable public network access for the public IP address of the machine (or its internet-facing proxy) that this is being run on.
+    .DESCRIPTION
+    This command updates a Key Vault to enable public network access for the public IP address of the machine (or its internet-facing proxy) that this is being run on. All existing IP address and VNet rules are maintained.
+    .PARAMETER SubscriptionId
+    The Azure subscription ID containing the Key Vault.
+    .PARAMETER ResourceGroupName
+    The Resource Group name containing the Key Vault.
+    .PARAMETER KeyVaultName
+    The Key Vault name.
+    .INPUTS
+    None
+    .OUTPUTS
+    None
+    .EXAMPLE
+    PS> Set-KeyVaultPublicNetworkAccessEnabledForMe -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -KeyVaultName "MyKeyVaultName"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $SubscriptionId,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $ResourceGroupName,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $KeyVaultName
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  # ##################################################
+  # Get my public IP address
+  $myPublicIpAddress = Invoke-RestMethod http://ipinfo.io/json | Select -exp ip
```

Comment:
In some env with constraint this public API might not be accessible, can we provide some alternative?

Reply:
Very valid concern. The issue is that if my machine doesn't have a local proxy config - e.g. if my local network sits behind infrastructure that does SNAT - I have to "go outside and look back" (so to speak) to see what my egress public IP is.

This site, ipinfo.io, is used in many serverfault and similar "how to" articles to solve this problem.

However, I added a check for whether this site can be connected to (HTTP 200) and if not, added an info msg to user saying cannot connect to public resource to get public IP address, no change to be made to Key Vault network access rules.